flow/connectors/clickhouse/normalize.go

```diff
@@ -266,19 +267,84 @@ func (c *ClickHouseConnector) NormalizeRecords(
 		return nil, fmt.Errorf("failed to copy avro stages to destination: %w", err)
 	}
 
+	rawTbl := c.getRawTableName(req.FlowJobName)
+	for batchID := normBatchID + 1; batchID <= req.SyncBatchID; batchID++ {
+		err = c.syncTablesInThisBatch(ctx, req, rawTbl, batchID)
+		if err != nil {
+			c.logger.Error("[clickhouse] error while syncing tables in this batch", slog.Any("error", err),
+				slog.Int64("batchID", batchID))
+			return nil, err
+		}
+
+		err = c.UpdateNormalizeBatchID(ctx, req.FlowJobName, batchID)
+		if err != nil {
+			c.logger.Error("[clickhouse] error while updating normalize batch id",
+				slog.Any("error", err),
+				slog.Int64("batchID", batchID))
+			return nil, err
+		}
+	}
+
+	return &model.NormalizeResponse{
+		Done:         true,
+		StartBatchID: normBatchID + 1,
+		EndBatchID:   req.SyncBatchID,
+	}, nil
+}
+
+func (c *ClickHouseConnector) getDistinctTableNamesInBatch(
+	ctx context.Context,
+	flowJobName string,
+	batchID int64,
+) ([]string, error) {
+	rawTbl := c.getRawTableName(flowJobName)
+
+	q := fmt.Sprintf(
+		`SELECT DISTINCT _peerdb_destination_table_name FROM %s WHERE _peerdb_batch_id = %d`,
```

Comment:
is it possible to craft a query to aggregate and fetch distinct tables for all batches at once

Reply:
That is the current behaviour and this PR changes it to get it for each batch
Do you want this function to be called higher up for all batches and pass a map of batchId:[table1,table2..] down to `syncTablesInThisBatch()`

Reply:
yes